core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -81,17 +99,29 @@ private[spark] class DiskBlockManager(conf: SparkConf, deleteFilesOnStop: Boolea
     new File(subDir, filename)
   }
 
-  def getFile(blockId: BlockId): File = getFile(blockId.name)
+  /**
+   * Used only for testing.
+   */
+  private[spark] def getFile(filename: String): File =
+    getFile(localDirs, subDirs, subDirsPerLocalDir, filename)
+
+  def getFile(blockId: BlockId): File = {
+    if (containerDirEnabled && blockId.isTemp) {
+      getFile(containerDirs, subContainerDirs, subDirsPerLocalDir, blockId.name)
+    } else {
+      getFile(localDirs, subDirs, subDirsPerLocalDir, blockId.name)
+    }
+  }
 
   /** Check if disk block manager has a block. */
   def containsBlock(blockId: BlockId): Boolean = {
-    getFile(blockId.name).exists()
```

Comment:
Is this a performance optimization? I guess we can leave the original style because `def getFile(blockId: BlockId): File` still exist.

Reply:
`def getFile(blockId: BlockId)` will check the blockId is temp block or not, if is a temp block, then storage to container directory.
`def getFile(filename: String)` just storage a block to local directory.

Looks like we can change `def getFile(filename: String)` to private since this PR changes all invoker outside class to `def getFile(blockId: BlockId)`:
```scala
-    val targetFile = diskManager.getFile(targetBlockId.name)
+   val targetFile = diskManager.getFile(targetBlockId)
```

Reply:
So `def getFile(filename: String)` is only used for testing now.

Reply:
I updated the description for the interface changing part.

Reply:
Now I refactor the test case and remove `def getFile(filename: String)` now!

Reply:
Thank you for explanation. It's a little confusing to me, but I'll take a look more in that way later.